### ISSUE TITLE
Update CRD to be Structural

### ISSUE THREAD

**Issue by cliveseldon** [SeldonIO/seldon-core#641]
https://kubernetes.io/blog/2019/06/20/crd-structural-schema/

**Comment by cliveseldon** [comment]
Worth waiting for k8s version to become widelyused before updating to v1beta1 of apiextensions

```
h this we conclude the discussion of the structural schema in Kubernetes 1.15 and beyond. To sum up:

    structural schemas are optional in apiextensions.k8s.io/v1beta1. Non-structural CRDs will keep working as before.

```

**Comment by cliveseldon** [comment]
Can only be fully structural when Kubernetes bug in #926 is fixed in Kubernetes 1.17
